### PR TITLE
CPLP-1248 app notification

### PR DESCRIPTION
**Pull request by Phil91**
In this PR Task [CPLP-1248](https://jira.catena-x.net/browse/CPLP-1248) is resolved

### CODE REVIEW

**Review by evegufy — APPROVED** [review]

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs [hunk 1]
```diff
@@ -285,4 +293,34 @@ private async Task<Guid> CreateAppAsync(AppRequestModel appRequestModel)
...
     }
+    
+    private async Task<(Guid companyId, CompanyAssignedApp? companyAssignedApp, string companyName)> GetCompanyAppSubscriptionData(Guid appId, string iamUserId, Guid requesterId)
+    {
+        var companyAssignedAppRepository = _portalRepositories.GetInstance<ICompanyAssignedAppsRepository>();
+        var companyAppSubscriptionData = await companyAssignedAppRepository
+            .GetCompanyIdWithAssignedAppForCompanyUserAsync(appId, iamUserId).ConfigureAwait(false);
+        if (companyAppSubscriptionData == default)
+        {
+            throw new ArgumentException($"user {iamUserId} is not assigned with a company");
```
Comment (ntruchsess) [inline src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs:304]: use ControllerArgumentException instead

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs [hunk 2]
```diff
@@ -168,24 +163,36 @@ public async Task AddOwnCompanyAppSubscriptionAsync(Guid appId, string iamUserId
...
 
-        if (subscription == null || subscription.AppSubscriptionStatusId != PortalBackend.PortalEntities.Enums.AppSubscriptionStatusId.PENDING)
+        if (subscription is not {AppSubscriptionStatusId: AppSubscriptionStatusId.PENDING})
         {
             throw new ArgumentException("No pending subscription for provided parameters existing.");
         }
-        subscription.AppSubscriptionStatusId = PortalBackend.PortalEntities.Enums.AppSubscriptionStatusId.ACTIVE;
+        subscription.AppSubscriptionStatusId = AppSubscriptionStatusId.ACTIVE;
+
+        var companyUserId = await _portalRepositories.GetInstance<IUserRepository>()
```
Comment (ntruchsess) [inline src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs:183]: To execute a single query the retrival of companyUserId should be included in above query (that comes for free as above query does join the companyUsers-table)

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs [hunk 3]
```diff
@@ -116,31 +118,10 @@ public async Task AddOwnCompanyAppSubscriptionAsync(Guid appId, string iamUserId
...
-                throw new ArgumentException($"company {companyId} is already subscribed to {appId}");
-            }
-            companyAssignedApp.AppSubscriptionStatusId = AppSubscriptionStatusId.PENDING;
-        }
-
-        await _portalRepositories.SaveAsync().ConfigureAwait(false);
+        
+        var (requesterId, requesterEmail) = await _portalRepositories.GetInstance<IUserRepository>()
+            .GetCompanyUserIdAndEmailForIamUserUntrackedAsync(iamUserId).ConfigureAwait(false);
+        var (_, _, companyName) = await GetCompanyAppSubscriptionData(appId, iamUserId, requesterId);
```
Comment (ntruchsess) [inline src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs:124]: please add a new query that does not return the first two items which are ignored here

**Review by ntruchsess — DISMISSED** [review]

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs (hunk 3 above)
Comment (Phil91) [reply src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs]: done

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs (hunk 1 above)
Comment (Phil91) [reply src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs]: changed

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs (hunk 2 above)
Comment (Phil91) [reply src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs]: done

**Comment by evegufy** [comment]
@Phil91 Go ahead with merging. The failed test coverage scan is ignored for now, as not representative.

**Pull request merged**